packages/statemanager/src/ethersStateManager.ts

```diff
@@ -251,23 +205,15 @@ export class EthersStateManager extends BaseStateManager implements StateManager
    * Both are represented as `0x` prefixed hex strings.
    */
   dumpStorage(address: Address): Promise<StorageDump> {
-    return new Promise((resolve, reject) => {
-      this._getStorageTrie(address)
-        .then((trie) => {
-          const storage: StorageDump = {}
-          const stream = trie.createReadStream()
-
-          stream.on('data', (val: any) => {
-            storage['0x' + val.key.toString('hex')] = '0x' + val.value.toString('hex')
-          })
-          stream.on('end', () => {
-            resolve(storage)
-          })
-        })
-        .catch((e) => {
-          reject(e)
-        })
-    })
+    const addressStorage = this.storageCache.get(address.toString())
+    if (addressStorage === undefined) {
+      return Promise.resolve({} as StorageDump)
+    }
+    const dump: StorageDump = {}
+    for (const slot of addressStorage) {
+      dump[slot[0]] = bufferToHex(slot[1])
+    }
```

Comment:
```suggestion
    const dump: StorageDump = {}
    if (addressStorage !== undefined) {
      for (const slot of addressStorage) {
        dump[slot[0]] = bufferToHex(slot[1])
      }
    }
```
slightly simpler imo